ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -795,7 +806,8 @@ public void alterTable(String catName, String dbName, String tblName, Table newT
       // Take a table snapshot and set it to newTbl.
       AcidUtils.TableSnapshot tableSnapshot = null;
       if (transactional) {
-        if (replWriteId > 0) {
+
+        if (inReplication(newTbl) && replWriteId > 0) {
```

Comment:
Why this change is needed?

Reply:
Rebased with master and only kept relevant changes.